### PR TITLE
update tinycbor to latest version

### PR DESCRIPTION
**Pull request by qiutongs**
Description
-----------

Update tinycbor (an unknown old version in 2016) to latest one: https://github.com/intel/tinycbor

I copy all the files in https://github.com/intel/tinycbor/tree/master/src

I test on windows desktop:
- FULL_SHADOW
- FULL_OTA_CBOR
- FULL_OTA_AGENT
- FULL_OTA_PAL

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

### CODE REVIEW

**Comment by sarenameas** [comment]
We need to run the tests on all boards that support OTA. As I recall, we have made our own custom updates for our set of supported compilers in lib\third_party\tinycbor\compilersupport_p.h.

File: lib/third_party/tinycbor/compilersupport_p.h [hunk 1]
```diff
@@ -77,14 +80,18 @@
...
+#endif
+
 #ifndef __has_builtin
 #  define __has_builtin(x)  0
 #endif
 
-
-/* Disable this optimization for TI ARM compiler v18 or higher because it has some issues
-with these intrinsics. */
-#if !defined(__TI_COMPILER_VERSION__) || __TI_COMPILER_VERSION__ < 18000000
```
Comment (sarenameas) [inline lib/third_party/tinycbor/compilersupport_p.h]: Deleting this intrinsic might cause issues on the TI platform so we should definitely run tests on TI (and all other platforms).

Comment (qiutongs) [reply lib/third_party/tinycbor/compilersupport_p.h]: Good catch. I blindly deleted all the old files. Let me try to run OTA E2E test on TI and see what happens

**Pull request closed without merging**